crates/core/src/rpc/surfnet_cheatcodes.rs

```diff
@@ -743,4 +754,82 @@ impl SvmTricksRpc for SurfnetCheatcodesRpc {
             })
         })
     }
+
+    fn get_transaction_profile(
+        &self,
+        meta: Self::Metadata,
+        transaction_signature: UuidOrSignature,
+    ) -> BoxFuture<Result<RpcResponse<ProfileResult>>> {
+        let svm_locker = match meta.get_svm_locker() {
+            Ok(locker) => locker,
+            Err(e) => return e.into(),
+        };
+        let first_slot = svm_locker
+            .get_first_local_slot()
+            .unwrap_or_else(|| svm_locker.with_svm_reader(|svm| svm.get_latest_absolute_slot()));
+        let latest_slot = svm_locker.with_svm_reader(|svm| svm.get_latest_absolute_slot());
+
+        Box::pin(async move {
+            let mut profile_result =
+                svm_locker.with_svm_reader(|svm| match &transaction_signature {
```

Comment:
Rather than `.with_svm_reader`, can we add a function to the locker to `get_transaction_profile`?